### PR TITLE
[java] Update security rules for pmd7

### PR DESCRIPTION
**Pull request by adangel**
## Describe the PR

* Updated both rules HardCodedCryptoKey and InsecureCryptoIv
* Enabled the unit tests
* Introduced a common package-private base class, since both rules do actually the same - checking for hard coded constructor arguments

## Related issues

- Part of #2701 

## Ready?

- [ ] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Complete build `./mvnw clean verify` passes (checked automatically by travis)
- [ ] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Running pmdtester failed, this message is mainly used to remind the maintainers of PMD.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by adangel** [comment]
TODO:
- [ ] need to enable the rules for regression tester -> https://github.com/pmd/pmd/blob/pmd/7.0.x/.ci/files/all-java.xml
- [ ] Figure out, what's wrong with pmd-regression-tester not finding the ruleset...

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/types/internal/infer/ast/CtorInvocMirror.java [hunk 1]
```diff
@@ -92,11 +92,16 @@ public TypeSpecies getStandaloneSpecies() {
...
+        if (typeMirror instanceof JClassType) {
+            JClassType classTypeMirror = (JClassType) typeMirror;
+            if (isDiamond()) {
+                classTypeMirror = classTypeMirror.withTypeArguments(classTypeMirror.getFormalTypeParams());
+            }
+            return classTypeMirror;
         }
-        return typeMirror;
+        // this might happen if the type is not known (e.g. SentinelType)
+        return null;
```
[the remark this answers is not in the document]
Comment (oowekyala) [reply pmd-java/src/main/java/net/sourceforge/pmd/lang/java/types/internal/infer/ast/CtorInvocMirror.java:104]: Looking at usages of the super method, it's easier to make this method return non-null by making up an unresolved symbol, than to adapt the call sites to a nullable return type.

[the remark this answers is not in the document]
Comment (oowekyala) [reply pmd-java/src/main/java/net/sourceforge/pmd/lang/java/types/internal/infer/ast/CtorInvocMirror.java]: Actually I kept a non-null return type but changed it to JTypeMirror, which is fine for most use sites. If it's unresolved, it returns the `UNKNOWN` type

**Pull request merged**